conan.cmake

```diff
@@ -83,6 +84,22 @@ function(conan_cmake_settings result)
 endfunction()
 
 
+function(conan_cmake_detect_gnu_libcxx result)
+    get_directory_property(defines DIRECTORY ${CMAKE_SOURCE_DIR} COMPILE_DEFINITIONS)
+    foreach(define ${defines})
+        if(define STREQUAL "_GLIBCXX_USE_CXX11_ABI=0")
+            set(${result} libstdc++ PARENT_SCOPE)
+            return()
+        endif()
+    endforeach()
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5.1")
```

Comment:
In any case, your approach is much better than the existing, so lets merge it, thanks very much!